core/api/monitor/v1/monitor_proto/monitor.proto

```diff
@@ -0,0 +1,102 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+option go_package = "github.com/google/keytransparency/core/api/monitor/v1/monitor_proto";
+
+// Monitor Service
+//
+// The Key Transparency monitor server service consists of APIs to fetch
+// monitor results queried using the mutations API.
+package google.keytransparency.monitor.v1;
+
+import "google/api/annotations.proto";
+import "api/v1/keytransparency_proto/keytransparency.proto";
+import "trillian.proto";
+
+// GetMonitoringRequest contains the input parameters of the GetMonitoring APIs.
+message GetMonitoringRequest {
+  // kt_url is the URL of the keytransparency server for which the monitoring
+  // result will be returned.
+  string kt_url = 2;
+
+  // domain_id identifies the merkel tree being monitored.
+  string domain_id = 3;
+
+  // epoch specifies the revision for which the monitoring results will
+  // be returned (epochs start at 0).
+  int64 epoch = 1;
+}
+
+message GetMonitoringResponse {
+  // smr contains the map root for the sparse Merkle Tree signed with the
+  // monitor's key on success. If the checks were not successful the
+  // smr will be empty. The epochs are encoded into the smr map_revision.
+  trillian.SignedMapRoot smr = 1;
+
+  // seen_timestamp_nanos contains the time in nanoseconds where this particular
+  // signed map root was retrieved and processed. The actual timestamp of the
+  // smr returned by the server is contained in above smr field.
+  int64 seen_timestamp_nanos = 2;
+
+  //
+  // The following fields provide more information about each failure in this
+  // response, if any.
+  //
+
+  // errors contains a list of error string representing the verification checks
+  // that failed while monitoring the key-transparency server.
+  repeated string errors = 3;
+
+  // error_data contains the original response from the mutations API if and
+  // only if at least one verification step failed. It can be used to re-run the
+  // verification steps.
+  keytransparency.v1.GetMutationsResponse error_data = 4;
+ }
+
+// The Monitor Service API allows clients to query the monitors observed and
+// validated signed map roots.
+//
+// - Signed Map Roots can be collected using the GetSignedMapRoot APIs.
+// - Monitor resources are named:
+//   - /monitor/v1/{kt-url}/domains/{domain_id}/revisions/{epoch}
+//   - /monitor/v1/{kt-url}/domains/{domain_id}/revision:latest
+//
+service MonitorService {
+  // GetSignedMapRoot returns the latest valid signed map root the monitor
+  // observed. Additionally, the response contains additional data necessary to
```

Comment:
Additionally ... additional
Maybe 'Additionally, the response contains extra data necessary...'

Reply:
thanks